src/core/ext/transport/cronet/transport/cronet_transport.cc

```diff
@@ -214,6 +216,26 @@ void grpc_cronet_stream_unref(stream_obj* s) { grpc_stream_unref(s->refcount); }
 
 static enum e_op_result execute_stream_op(struct op_and_state* oas);
 
+static const size_t tail_xtra[4] = {0, 0, 1, 2};
```

Comment:
Is this code (infering length) copied from chttps? Any way to avoid duplication?

Reply:
No, chttp does not do this explicitly. However I do think this code should be in `bin_decoder`.